tools/crd-client-generator-js/src/generateClient.ts

```diff
@@ -1,59 +1,90 @@
-const YAML = require('yaml');
-const fs = require('fs');
-import Generator from 'openapi-to-flowtype/dist/Generator';
-const prettier = require('prettier');
+const YAML = require("yaml");
+const fs = require("fs");
+import Generator from "openapi-to-flowtype/dist/Generator";
+const prettier = require("prettier");
 
 export const DEFAULT_PRETTIER_OPTIONS = {
-  parser: 'babel',
+  parser: "babel",
 };
 
-export function generateClient(crdFile: string, destinationFile: string, prefix?: string) {
+export function generateClient(
+  crdFile: string,
+  destinationFile: string,
+  prefix?: string
+) {
   const prettierOptions = {
     ...DEFAULT_PRETTIER_OPTIONS,
-    ...( prettier.resolveConfig.sync( destinationFile ) || {} )
+    ...(prettier.resolveConfig.sync(destinationFile) || {}),
   };
-  
-  const file = fs.readFileSync(crdFile, 'utf8');
+
+  const file = fs.readFileSync(crdFile, "utf8");
   const crdSpec = YAML.parse(file);
-  
+
   const generator = new Generator();
+
+  const versions =
+    crdSpec.apiVersion === "apiextensions.k8s.io/v1"
+      ? crdSpec.spec.versions.map((version) => version.name)
+      : [crdSpec.spec.version];
```

Comment:
Sorry, just caught this: I don't think the `versions` field only exists in `v1`. It's the opposite: in `v1beta1`, there is **also** a `version` field, and both may be specified (in which case `version` **must** match `versions[0]`).